### PR TITLE
Mechanism base

### PR DESCRIPTION
**Pull request by erendrake**
This has a bunch of fixes  as well as some refactoring and cleanup work

### CODE REVIEW

File: InfernalRobotics/InfernalRobotics/Control/Servo/MechanismBase.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
+using System;
```
Comment (erendrake) [inline InfernalRobotics/InfernalRobotics/Control/Servo/MechanismBase.cs]: almost everything in rotating and translating are the same so i gave them a base class :)

File: InfernalRobotics/InfernalRobotics/Control/Servo/TranslateMechanism.cs [hunk 2]
```diff
@@ -1,136 +1,32 @@
...
             set
             {
-                rawServo.speedTweak = Math.Max(value, 0.01f);
+                var clamped = Mathf.Clamp(value, RawServo.translateMin, RawServo.translateMax);
+                RawServo.minTweak = clamped;
             }
         }
 
-        public float AccelerationLimit
+        public override float MaxPositionLimit
```
Comment (erendrake) [inline InfernalRobotics/InfernalRobotics/Control/Servo/TranslateMechanism.cs]: max used to be clamping to the rotation extents, fixed

File: InfernalRobotics/InfernalRobotics/Utility/TextureLoader.cs [hunk 3]
```diff
@@ -9,31 +7,30 @@ namespace InfernalRobotics.Utility
 {
     public class TextureLoader
     {
-        private static bool isReady = false;
```
Comment (erendrake) [inline InfernalRobotics/InfernalRobotics/Utility/TextureLoader.cs]: almost everything in here is wrapping up the textures in a property, useKSPAssets was not being used.

Comment (ZiwKerman) [inline InfernalRobotics/InfernalRobotics/Utility/TextureLoader.cs]: I was going to add an option to load icons from GameAssets. But it is such a small feature, can be fixed later

**Comment by ZiwKerman** [comment]
I was actually going to suggest that those 2 classes had too much in common. I see no problems here.

**Pull request merged**